android/java/org/chromium/chrome/browser/brave_leo/BraveLeoPrefUtils.java

```diff
@@ -0,0 +1,59 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+package org.chromium.chrome.browser.brave_leo;
+
+import org.chromium.base.ContextUtils;
+import org.chromium.base.Log;
+import org.chromium.chrome.browser.app.BraveActivity;
+import org.chromium.chrome.browser.preferences.BravePref;
+import org.chromium.chrome.browser.profiles.Profile;
+import org.chromium.components.user_prefs.UserPrefs;
+
+public class BraveLeoPrefUtils {
+    private static final String TAG = "BraveLeoPrefUtils";
+
+    private static Profile getProfile() {
+        Profile profile = null;
+        try {
+            BraveActivity activity = BraveActivity.getBraveActivity();
+            profile = activity.getCurrentProfile();
+        } catch (BraveActivity.BraveActivityNotFoundException e) {
+            Log.e(TAG, "get BraveActivity exception", e);
+        }
+        if (profile == null) {
+            Log.e(TAG, "BraveLeoPrefUtils.getProfile profile is null");
+        }
+
+        return profile;
+    }
+
+    public static void setIsSubscriptionActive(boolean value) {
+        UserPrefs.get(BraveLeoPrefUtils.getProfile())
```

Comment:
looks like `getProfile` can return `null`, shouldn't we check for that?

Reply:
It's a good point, I'm going to fix that on the upcoming PR rebase.

Reply:
done